src/ape/managers/converters.py

```diff
@@ -48,13 +48,18 @@ def convert(self, value: str) -> bytes:
 class HexIntConverter(ConverterAPI):
     """
     Convert hex values to integers.
+
+    **NOTE** If value is a ``str``, it must begin with "0x".
     """
 
     def is_convertible(self, value: Any) -> bool:
         return (isinstance(value, str) and is_hex(value)) or isinstance(value, bytes)
 
     def convert(self, value: Any) -> int:
-        return to_int(HexBytes(value))
+        if isinstance(value, bytes) or (value.startswith("0x") and isinstance(value, str)):
```

Comment:
start with checking if a str before checking if it starts with 0x or else you can get a type error

```suggestion
        if isinstance(value, bytes) or (isinstance(value, str) and value.startswith("0x")):
```